src/vs/workbench/contrib/notebook/browser/viewModel/baseCellViewModel.ts

```diff
@@ -104,10 +104,8 @@ export abstract class BaseCellViewModel extends Disposable {
 		return this._focusMode;
 	}
 	set focusMode(newMode: CellFocusMode) {
-		if (this._focusMode !== newMode) {
-			this._focusMode = newMode;
-			this._onDidChangeState.fire({ focusModeChanged: true });
-		}
+		this._focusMode = newMode;
+		this._onDidChangeState.fire({ focusModeChanged: true });
```

Comment:
Reading through the code again, it seems that we used to set `focusMode` to trigger the cell editor to be focused again, for example, https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/notebook/browser/notebookEditorWidget.ts#L1767

To avoid depending on the event listener, what we should probably do here is find the right code editor through `_renderedEditors` and then do a `editor.focus()`.

Reply:
Exactly, we depend on the side effect. I forgot that we have `_renderedEditors`, using that would be a better solution. So you aren't worried about this change breaking something?